programs/perpetuals/tests/anchor/cryptography.ts

```diff
@@ -0,0 +1,22 @@
+import * as sha3 from "sha3";
+import { randomBytes } from "crypto";
+import * as secp256k1 from "secp256k1";
+
+export function generateSecp256k1Keypair() {
+    // Private key for secp256k1 is just random bytes. Public key is derived afterwards.
+    let privateKey;
+    do {
+      privateKey = randomBytes(32)
+    } while (!secp256k1.privateKeyVerify(privateKey));
+    // Get the public key in uncompressed format.
+    const publicKey = secp256k1.publicKeyCreate(privateKey, false);
+    // The library always returns a 65 byte public key that starts with 0x04, so omit the first byte.
+    return [privateKey, publicKey.slice(1)];
+}
+
+export function secp256k1Sign(privateKey, message) {
```

Comment:
can also straight sign with the anchor namespace .signer() command or along those lines

Reply:
it seems those were for transactions;  went with TweetNaCl per the cookbook https://solanacookbook.com/references/keypairs-and-wallets.html#how-to-sign-and-verify-messages-with-wallets  please let me know if there is a better approach!